test/CMakeLists.txt

```diff
@@ -11,5 +13,14 @@ add_executable(menoh_test
     #vgg16.cpp
 
 )
-target_link_libraries(menoh_test gtest_main menoh)
+target_link_libraries(menoh_test gtest_main menoh_test_target)
+
+if(NOT LINK_STATIC_LIBPROTOBUF)
+    if(DEFINED PROTOBUF_LIBRARIES)
+        target_link_libraries(menoh_test ${PROTOBUF_LIBRARIES})
+    else()
+        message(FATAL_ERROR "PROTOBUF_LIBRARIES is not found")
+    endif()
+endif()
```

Comment:
I cannot understand why directly linking `protobuf` is necessary.
`menoh_test` seems not using `protobuf` directly and  I expect it works without linking protobuf directly, in particular if `LINK_STATIC_LIBPROTOBUF=OFF`.

Reply:
Same as https://github.com/pfnet-research/menoh/pull/70#discussion_r213889756.